src/lib/core/style-builder/style-builder.ts

```diff
@@ -10,5 +10,14 @@ import {StyleDefinition} from '../style-utils/style-utils';
 
 @Injectable()
 export abstract class StyleBuilder {
-  abstract buildStyles(input: string, parent?: Object): StyleDefinition;
+  abstract buildStyles(input: string, parent?: Object): StyleBuilderOutput;
+
+  sideEffect(_input: string, _styles: StyleDefinition, _parent?: Object) {
```

Comment:
What is the intent of the `sideEffect`; should we document it here with comments?

Reply:
The intent is to have some routine execute regardless of caching, with the result of the computed styles, but totally opt-in. I'll add a small comment for it